core/src/com/group66/game/screens/SettingsScreen.java

```diff
@@ -2,89 +2,91 @@
 package com.group66.game.screens;
 
 import com.badlogic.gdx.Gdx;
-import com.badlogic.gdx.Screen;
-import com.badlogic.gdx.graphics.Color;
 import com.badlogic.gdx.graphics.GL20;
-import com.badlogic.gdx.graphics.Pixmap;
-import com.badlogic.gdx.graphics.Pixmap.Format;
-import com.badlogic.gdx.graphics.Texture;
-import com.badlogic.gdx.graphics.g2d.BitmapFont;
 import com.badlogic.gdx.scenes.scene2d.Actor;
 import com.badlogic.gdx.scenes.scene2d.Stage;
-import com.badlogic.gdx.scenes.scene2d.ui.Skin;
 import com.badlogic.gdx.scenes.scene2d.ui.TextButton;
-import com.badlogic.gdx.scenes.scene2d.ui.TextButton.TextButtonStyle;
 import com.badlogic.gdx.scenes.scene2d.utils.ChangeListener;
 import com.group66.game.BustaMove;
-import com.group66.game.helpers.AssetLoader;
 import com.group66.game.helpers.DifficultyManager;
 import com.group66.game.logging.MessageType;
 import com.group66.game.settings.Config;
 
 /**
  * A Class for the SettingsScreen of the game.
  */
-public class SettingsScreen implements Screen {
+public class SettingsScreen extends AbstractMenuScreen {
 
-    private Stage stage;
-    private Skin skin;
```

Comment:
Disposing of the screens is working all well in all screens except of the shop screen (only when accessed between levels in the career mode, for some reason I don't know).
I have a dispose method in the abstract class, plus Skin and Stage objects are also declared in the AbstractGameClass.
Any elaboration on your suggestion?
and what exactly do you mean by "If you remove them here, they can't be disposed in de dispose() method"?